version: '3.9'

services:
  db-proyecto:
    container_name: db-proyecto
    hostname: db-proyecto
    image: mysql:debian
    restart: unless-stopped
    tty: true
    stdin_open: true
    ports:
      - "3306:3306"
    expose:
      - 3306
    cap_add:
      - SYS_NICE
    environment:
      MYSQL_ROOT_PASSWORD: root123
      MYSQL_LOG_CONSOLE: "true"
    volumes:
      - ./config/my.cnf:/etc/mysql/my.cnf
      - ./sql/scheme.sql:/docker-entrypoint-initdb.d/1.sql:ro
      - ./sql/indexes.sql:/docker-entrypoint-initdb.d/2.sql:ro
      - ./sql/values.sql:/docker-entrypoint-initdb.d/3.sql:ro
    command: bash -c "chmod 644 /etc/mysql/my.cnf && docker-entrypoint.sh mysqld"
    mem_limit: 512M
    networks:
      bridge-db:
        ipv4_address: 192.168.20.20
  
networks:
  bridge-db:
    name: bridge-db
    driver: bridge
    ipam:
      config:
        - subnet: 192.168.20.0/24
          gateway: 192.168.20.1